for review

SingeRoi opened the pull request:
for review

SingeRoi committed:
sets up allauth

SingeRoi committed:
adds allauth templates and base template

SingeRoi committed:
adds home app and templates

SingeRoi committed:
adds home text edit

SingeRoi committed:
adds media base css image links

SingeRoi committed:
Merge pull request #1 from SingeRoi/home

adds media base css image links

SingeRoi committed:
adds includes main_nav mobile_top_header

SingeRoi committed:
Merge pull request #2 from SingeRoi/home

adds includes main_nav mobile_top_header

SingeRoi committed:
adds product app, models and fixtures

SingeRoi committed:
Merge pull request #3 from SingeRoi/home

adds product app, models and fixtures

SingeRoi committed:
adds chapters, media datset, products templates,  migrates chapters, updates admin, settings, models, views

SingeRoi committed:
Merge pull request #4 from SingeRoi/home

adds chapters, media dataset, products templates,  migrates chapters, …

SingeRoi committed:
adds pagination to products

SingeRoi committed:
Merge pull request #5 from SingeRoi/home

adds pagination to products

SingeRoi committed:
adds functionality to navigation and filtering

SingeRoi committed:
adds to chapter model, fixes comics sorting

SingeRoi committed:
Merge pull request #6 from SingeRoi/home

Home

SingeRoi committed:
adds profile and bag apps, profile functionality

SingeRoi committed:
Merge pull request #7 from SingeRoi/home

adds profile and bag apps, profile functionality

SingeRoi committed:
adds products chapter css and js, adds registration functionality, updates models and migrates

SingeRoi committed:
Merge pull request #8 from SingeRoi/home

adds products chapter css and js, adds registration functionality, up…

SingeRoi committed:
updates admin format, updates responsive styling

SingeRoi committed:
restyles product page

SingeRoi committed:
Merge pull request #9 from SingeRoi/home

Home

SingeRoi committed:
adds toast htmls, adds fix for redundant chapter name, styles chapter display

SingeRoi committed:
Merge pull request #10 from SingeRoi/home

adds toast htmls, adds fix for redundant chapter name, styles chapter…

SingeRoi committed:
adds Product Admin -  Product Form

SingeRoi committed:
adds Product Admin - adds product view, url and template

SingeRoi committed:
adds profiles, templates urls, views

SingeRoi committed:
Merge pull request #11 from SingeRoi/home

Home

SingeRoi committed:
adds add_product function, minor styling to overlay

SingeRoi committed:
Merge pull request #12 from SingeRoi/home

adds add_product function, minor styling to overlay

SingeRoi committed:
Product admin - adds editing products, profile minor restyling

SingeRoi committed:
Merge pull request #13 from SingeRoi/home

Product admin - adds editing products, profile minor restyling

SingeRoi committed:
Product admin - adds delete products, adds edit delete buttons

SingeRoi committed:
Merge pull request #14 from SingeRoi/home

Product admin - adds delete products, adds edit delete buttons

SingeRoi committed:
fixes display of logout confirmation

SingeRoi committed:
fixes mobile display of logout confirmation, adds profile form link on mobile navbar

SingeRoi committed:
Merge pull request #15 from SingeRoi/home

Home

SingeRoi committed:
adds bookCoin form, view, adds + migrates modalcode, chapter btn styles

SingeRoi committed:
Merge pull request #16 from SingeRoi/home

adds bookCoin form, view, adds + migrates modalcode, chapter btn styles

SingeRoi committed:
adds cancel function to forms, minor restyle forms, fixes profile form link

SingeRoi committed:
Merge pull request #17 from SingeRoi/home

adds cancel function to forms, minor restyle forms, fixes profile for…

SingeRoi committed:
adds stripe

SingeRoi committed:
Merge pull request #18 from SingeRoi/work

adds stripe

SingeRoi committed:
update to stripe, updates fontawesome link, formats bookshelf, verfification message, rating display

SingeRoi committed:
Merge pull request #19 from SingeRoi/work

update to stripe, updates fontawesome link, formats bookshelf, verfif…

SingeRoi committed:
fixes display password reset pages, fixes mobile navbar display

SingeRoi committed:
Merge pull request #20 from SingeRoi/work

fixes display password reset pages, fixes mobile navbar display

SingeRoi committed:
adds user_rating input

SingeRoi committed:
Merge pull request #21 from SingeRoi/work

adds user_rating input

SingeRoi committed:
fixes display of bookshelf in profile,inc pyTest on products

SingeRoi committed:
Merge pull request #22 from SingeRoi/work

fixes display of bookshelf in profile,inc pyTest on products

ignatiusukwuoma commented on README.md:
```diff
@@ -0,0 +1,382 @@
+
+## Demo
+
+![Project Demo](/UXD/Demo.png)
```
If you intend to input an image from your repo here... just copy the url of the image in your repo and use it here

ignatiusukwuoma commented on README.md:
```diff
@@ -0,0 +1,382 @@
+
+### Wireframes
+ 
+[Here are the intitial wireframes](UXD)
```
Link to the folder on your repo

ignatiusukwuoma commented on README.md:
```diff
@@ -0,0 +1,382 @@
+
+    Heroku              https://testbookworm.herokuapp.com
+
+###Heroku Deployment 
```
add whitespace after the hashes, before Heroku

ignatiusukwuoma commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
You should add this file to gitignore and run `git rm -rf .idea/` to stop git from tracking it

ignatiusukwuoma commented on bag/urls.py:
```diff
@@ -0,0 +1,7 @@
+
+urlpatterns = [
+    path('', views.view_bag, name='view_bag'),
+    # path('add/<item_id>/', views.add_to_bag, name='add_to_bag'),
```
Remove commented code before you submit. I see the button that says `Add to Bag` for a title (not a chapter) doesn't do anything. Give user feedback if an action won't work as expected

ignatiusukwuoma commented on bookCoins/models.py:
```diff
@@ -0,0 +1,104 @@
+from django.db import models
```
duplicate import. delete this

ignatiusukwuoma commented on bookCoins/models.py:
```diff
@@ -0,0 +1,104 @@
+    county = models.CharField(max_length=80, null=True, blank=True)
+    postcode = models.CharField(max_length=20, null=True, blank=True)
+    country = CountryField(blank_label='Country *', null=False, blank=False)
+    date = models.DateTimeField(auto_now_add=True)
```
the field name - date can be confusing when it's actually a timestamp

ignatiusukwuoma commented on bookCoins/static/bookCoins/js/stripe_elements.js:
```diff
@@ -0,0 +1,137 @@
+});
+
+
+function update() {
```
Add comment for this function

ignatiusukwuoma commented on bookCoins/views.py:
```diff
@@ -0,0 +1,232 @@
+
+
+
+@csrf_exempt
```
why do you have csrf_exempt here?

ignatiusukwuoma commented on bookCoins/views.py:
```diff
@@ -0,0 +1,232 @@
+    """ Add a quantity of the specified product to the shopping bag """
+    redirect_url = request.POST.get('redirect_url')
+    current_coin = request.POST.get('current_coin')
+    if not request.POST.get('client_secret'):
+        pid={}
+    else:
+        pid = request.POST.get('client_secret').split('_secret')[0]
```
you should write this in this way:
```
pid={}
if request.POST.get('client_secret'):
  pid = request.POST.get('client_secret').split('_secret')[0]
```

ignatiusukwuoma commented on bookCoins/views.py:
```diff
@@ -0,0 +1,232 @@
+    return redirect(redirect_url)
+
+
+def topup_coins(request):
```
add docstring

ignatiusukwuoma reviewed (commented):
You need to improve the activation email message template.
The link sent in the activation email returns a Server Error (500)

SingeRoi commented:
Thanks I’ll look at that – i hadn’t seen that on my own tests but I’ll try it again maybe also from a different computer.
R

Provenance : Courrier<https://go.microsoft.com/fwlink/?LinkId=550986> pour Windows 10

De : Ignatius Ukwuoma<mailto:notifications@github.com>
Envoyé le :dimanche 4 octobre 2020 16:56
À : SingeRoi/bookWorm_ms4<mailto:bookWorm_ms4@noreply.github.com>
Cc : SingeRoi<mailto:bonjour_robert@outlook.com>; Author<mailto:author@noreply.github.com>
Objet :Re: [SingeRoi/bookWorm_ms4] for review (#23)


@ignatiusukwuoma commented on this pull request.

You need to improve the activation email message template.
The link sent in the activation email returns a Server Error (500)


In README.md<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499242220>:

> +## Site purpose

+

+This is a test of an English language webnovel publishing site of a type popular in China and of growing popularity elsewhere. The owner's purpose is to provide a source of new eNovels and eManga as well as other online publications that will be if interest to readers. The owner will gather income from purchase of access to publications and individual chapters are usually provided as the main products. Titles are often published in a serialised form with readers awaiting the next installment. Typical access is via mobile devices. The site provides a basic reader function to access content linked to registered user accounts. Some content is normally provided for free as a taster and the site free content is available to any reader.

+

+The owner also wishes to recruit writers attracted to contribute their work to the genres onsite to become providers of future content as well as translators to provide English and other language content. These users will be paid a fee for their work on paid content.

+

+Users purchase titles by purchasing credit in their profile on the site using a card (in reality usually methods such as WeChat or Alipay dominate). These then allow purchase of chapters. The presence of credit encourages users to keep returning to the site. Other payment schemes such as subscriptions exist but these are outside the scope of this demo project

+

+

+## About bookWorm.

+

+Site users can browse titles according to type (book, comic) or genre (sci-fi, fantasy or historical). After registration and login users can read sample chapters and then choose to purchase further chapters or a subscription. Users can then review the products they buy. Users can also submit their own works for online publication and/or may offer to act as translators for new works users will receive payment for this.

+

+## Demo

+

+![Project Demo](/UXD/Demo.png)

If you intend to input an image from your repo here... just copy the url of the image in your repo and use it here


In README.md<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499242399>:

> +| #694F5D       | 105, 79, 93 | eggplant|

+| #000000     | 0, 0, 0 | black |

+

+

+

+

+#### Typography

+Fonts chosen were

+

+#### Icons

+

+I used [Font Awesome](https://fontawesome.com/icons?d=gallery) for the icons in this project

+

+### Wireframes

+

+[Here are the intitial wireframes](UXD)

Link to the folder on your repo


In README.md<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499242737>:

> +#### Python

+

+[PEP8]( http://pep8online.com/)

+

+

+## Deployment

+

+Deployment and version control was carried out using GitHub and Heroku. The repository location is as follows:

+

+the sites are;

+

+    Github repository   https://github.com/SingeRoi/bookWorm_ms4

+

+    Heroku              https://testbookworm.herokuapp.com

+

+###Heroku Deployment

add whitespace after the hashes, before Heroku


In .idea/vcs.xml<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499242878>:

> @@ -0,0 +1,6 @@

+<?xml version="1.0" encoding="UTF-8"?>

+<project version="4">

+  <component name="VcsDirectoryMappings">

+    <mapping directory="" vcs="Git" />

+  </component>

+</project>

You should add this file to gitignore and run git rm -rf .idea/ to stop git from tracking it


In bag/urls.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499243448>:

> @@ -0,0 +1,7 @@

+from django.urls import path

+from . import views

+

+urlpatterns = [

+    path('', views.view_bag, name='view_bag'),

+    # path('add/<item_id>/', views.add_to_bag, name='add_to_bag'),

Remove commented code before you submit. I see the button that says Add to Bag for a title (not a chapter) doesn't do anything. Give user feedback if an action won't work as expected


In bookCoins/models.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499244773>:

> @@ -0,0 +1,104 @@

+from django.db import models

duplicate import. delete this


In bookCoins/models.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499245436>:

> +

+

+class OrderCoin(models.Model):

+    order_number = models.CharField(max_length=32, null=False, editable=False)

+    user_profile = models.ForeignKey(UserProfile, on_delete=models.SET_NULL,

+                                     null=True, blank=True, related_name='coin_orders')

+    full_name = models.CharField(max_length=50, null=False, blank=False)

+    email = models.EmailField(max_length=254, null=False, blank=False)

+    phone_number = models.CharField(max_length=20, null=False, blank=False)

+    street_address1 = models.CharField(max_length=80, null=False, blank=False)

+    street_address2 = models.CharField(max_length=80, null=True, blank=True)

+    town_or_city = models.CharField(max_length=40, null=False, blank=False)

+    county = models.CharField(max_length=80, null=True, blank=True)

+    postcode = models.CharField(max_length=20, null=True, blank=True)

+    country = CountryField(blank_label='Country *', null=False, blank=False)

+    date = models.DateTimeField(auto_now_add=True)

the field name - date can be confusing when it's actually a timestamp


In bookCoins/static/bookCoins/js/stripe_elements.js<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499253607>:

> +                card.update({ 'disabled': false});

+                $('#submit-button').attr('disabled', false);

+            } else {

+                if (result.paymentIntent.status === 'succeeded') {

+                    form.submit();

+                }

+            }

+        });

+    }).fail(function () {

+        // just reload the page, the error will be in django messages

+        location.reload();

+    })

+});

+

+

+function update() {

Add comment for this function


In bookCoins/views.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499253855>:

> +    try:

+        pid = request.POST.get('client_secret').split('_secret')[0]

+        stripe.api_key = settings.STRIPE_SECRET_KEY

+        stripe.PaymentIntent.modify(pid, metadata={

+            'save_info': request.POST.get('save_info'),

+            'username': request.user,

+        })

+        return HttpResponse(status=200)

+    except Exception as e:

+        messages.error(request, 'Sorry, your payment cannot be \

+            processed right now. Please try again later.')

+        return HttpResponse(content=e, status=400)

+

+

+

+@csrf_exempt

why do you have csrf_exempt here?


In bookCoins/views.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499254112>:

> +    if not request.POST.get('client_secret'):

+        pid={}

+    else:

+        pid = request.POST.get('client_secret').split('_secret')[0]

you should write this in this way:

pid={}

if request.POST.get('client_secret'):

  pid = request.POST.get('client_secret').split('_secret')[0]


In bookCoins/views.py<https://github.com/SingeRoi/bookWorm_ms4/pull/23#discussion_r499254252>:

> +def update_coins(request):

+    """ Add a quantity of the specified product to the shopping bag """

+    redirect_url = request.POST.get('redirect_url')

+    current_coin = request.POST.get('current_coin')

+    if not request.POST.get('client_secret'):

+        pid={}

+    else:

+        pid = request.POST.get('client_secret').split('_secret')[0]

+    request.session['current_coin'] = current_coin

+    request.session['pid'] = pid

+    request.session.modified = True

+

+    return redirect(redirect_url)

+

+

+def topup_coins(request):

add docstring

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/SingeRoi/bookWorm_ms4/pull/23#pullrequestreview-501619663>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALLX34AE32FLRQ7TM55JXJ3SJCEIZANCNFSM4SDTYJZA>.

SingeRoi committed:
adds userrating functionality updates stripe styles myprofile, deletes bag app

SingeRoi committed:
Merge pull request #24 from SingeRoi/work

adds userrating functionality updates stripe styles myprofile, delete…

SingeRoi committed:
README edits, adds ERD and images

SingeRoi committed:
Merge pull request #25 from SingeRoi/work

README edits, adds ERD and images

SingeRoi committed:
product detail.html; fixed buy chapter; update.js, chapter.js, rating.js

SingeRoi committed:
Merge pull request #26 from SingeRoi/work

product detail.html; fixed buy chapter; update.js, chapter.js, rating.js

SingeRoi committed:
new heroku deploy

SingeRoi committed:
remove cloud

SingeRoi committed:
Merge pull request #27 from SingeRoi/work

Work

SingeRoi committed:
allowedhost changed

SingeRoi committed:
Merge pull request #28 from SingeRoi/work

allowedhost changed

SingeRoi committed:
fix requirements

SingeRoi committed:
readme updates wireframes gif

SingeRoi committed:
updates image links

SingeRoi committed:
updates image links
